db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/StoreDataManagerProvider.java

```diff
@@ -15,42 +15,37 @@
  */
 package org.commonjava.indy.infinispan.data;
 
-import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.cassandra.data.ClusterStoreDataManager;
+import org.commonjava.indy.core.conf.IndyDurableStateConfig;
 import org.commonjava.indy.data.StandaloneStoreDataManager;
 import org.commonjava.indy.data.StoreDataManager;
 
-import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 import javax.enterprise.inject.Default;
 import javax.enterprise.inject.Produces;
 import javax.inject.Inject;
 
 @ApplicationScoped
-public class StoreDataManagerProvider {
+public class StoreDataManagerProvider
+{
 
     @Inject
-    @StandaloneStoreDataManager
-    InfinispanStoreDataManager infinispanStoreDataManager;
-
-    @Inject
-    IndyConfiguration indyConfiguration;
-
-    private StoreDataManager storeDataManager;
-
-    private boolean isStandaloneMode;
-
-    @PostConstruct
-    public void init()
-    {
-        isStandaloneMode = indyConfiguration.isStoreManagerStandalone();
-    }
+    IndyDurableStateConfig durableStateConfig;
 
     @Produces
     @Default
-    public StoreDataManager getStoreDataManager()
+    public StoreDataManager getStoreDataManager(
+                    @StandaloneStoreDataManager InfinispanStoreDataManager ispnStoreDataManager,
+                    @ClusterStoreDataManager StoreDataManager clusterStoreDataManager )
     {
-        // If isStandaloneMode is false, return CassandraStoreDataManager instantiation
-        storeDataManager = isStandaloneMode ? infinispanStoreDataManager : infinispanStoreDataManager;
-        return storeDataManager;
+        if ( durableStateConfig.getStoreStorage().equals( IndyDurableStateConfig.STORAGE_INFINISPAN ) )
+        {
+            return ispnStoreDataManager;
+        }
+        else if ( durableStateConfig.getStoreStorage().equals( IndyDurableStateConfig.STORAGE_CASSANDRA ) )
+        {
+            return clusterStoreDataManager;
+        }
+        return null;
```

Comment:
Consider not return null here but a default one, like ispn one.(Or memory one)

Reply:
And I'm not sure if it is better to use @Default and @Alternative way to control this inject.

Reply:
That would be an ideal way for this case, but we want to support to inject the specific implementation according to the config, in case that we can switch off easily without code change.

Reply:
> Consider not return null here but a default one, like ispn one.(Or memory one)

Yeah, that's a good idea.

Reply:
It should raise a runtime exception when the configuration is valid as John mentioned in the issue:
`No default values, to avoid accidentally using something that won't be durable`
I will update it later.